wpa_supplicant/build.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+set -e
+
+WPA_SUPPLICANT=wpa_supplicant-2.9
+
+b_log "Building wpa_supplicant"
+PREFIX_WPA_SUPPLICANT="${TOPDIR}/phoenix-rtos-ports/wpa_supplicant"
+PREFIX_WPA_SUPPLICANT_BUILD="${PREFIX_BUILD}/wpa_supplicant"
+PREFIX_WPA_SUPPLICANT_SRC="${PREFIX_BUILD}/wpa_supplicant/${WPA_SUPPLICANT}"
+PREFIX_WPA_SUPPLICANT_MARKERS="${PREFIX_WPA_SUPPLICANT_BUILD}/markers"
+PREFIX_WPA_SUPPLICANT_INSTALL="${PREFIX_WPA_SUPPLICANT_BUILD}/install"
+
+#
+# Download and unpack
+#
+mkdir -p "$PREFIX_WPA_SUPPLICANT_BUILD"
+
+if [ -n "$CLEAN" ]; then
+	rm -rf "${PREFIX_WPA_SUPPLICANT_BUILD:?}/*"
+fi
+
+[ -f "${PREFIX_WPA_SUPPLICANT}/${WPA_SUPPLICANT}.tar.gz" ] || wget https://w1.fi/releases/${WPA_SUPPLICANT}.tar.gz -P "$PREFIX_WPA_SUPPLICANT" --no-check-certificate
+[ -d "$PREFIX_WPA_SUPPLICANT_SRC" ] || tar zxf "${PREFIX_WPA_SUPPLICANT}/${WPA_SUPPLICANT}.tar.gz" -C "$PREFIX_WPA_SUPPLICANT_BUILD"
+
+#
+# Apply patches
+#
+mkdir -p "$PREFIX_WPA_SUPPLICANT_MARKERS"
+
+for patchfile in "${PREFIX_WPA_SUPPLICANT}"/patches/*.patch; do
+    if [ ! -f "${PREFIX_WPA_SUPPLICANT_MARKERS}/$(basename "$patchfile").applied" ]; then
+        echo "applying patch: $patchfile"
+        patch -d "$PREFIX_WPA_SUPPLICANT_SRC" -p1 < "$patchfile"
+        touch "${PREFIX_WPA_SUPPLICANT_MARKERS}/$(basename "$patchfile").applied"
+    fi
+done
+
+#
+# Make
+#
+mkdir -p "$PREFIX_WPA_SUPPLICANT_INSTALL"
+
+pushd "${PREFIX_WPA_SUPPLICANT_SRC}/wpa_supplicant"
+cp "$PREFIX_WPA_SUPPLICANT/.config" .config
+make install EXTRA_CFLAGS="-D__phoenix__" DESTDIR="$PREFIX_WPA_SUPPLICANT_INSTALL" LIBDIR="/lib" INCDIR="/include" BINDIR="/sbin"
+popd
+
+mkdir -p "$PREFIX_PROG" && cp -Ra "${PREFIX_WPA_SUPPLICANT_INSTALL}/sbin/wpa_cli" "${PREFIX_PROG}/wpa_cli"
```

Comment:
no need for `mkdir`, they are created by root `build.sh` 

`cp -a`, `-R` is redundant